ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1505,24 +1505,40 @@ static const char* GetRelativePath(const char* fullPath, const char* root)
         return "";
     }
     
+    string altPath(fullPath);
+    string altRoot(root);
+    const char* removePart = "/System/Volumes/Data";
+    const char* realFullPath;
+    if (altPath.find(removePart) == 0 && altRoot.find(removePart) != 0)
+    {
+        // If fullPath contains "/System/Volumes/Data", but the root does not remote it for comparison
+        // "/System/Volumes/Data/Users/account" is effectively "/Users/account" and should be compared as such
+        realFullPath = fullPath + strlen(removePart);
+    }
+    else
+    {
+        realFullPath = fullPath;
+    }
+
     size_t rootLength = strlen(root);
-    size_t pathLength = strlen(fullPath);
-    if (pathLength < rootLength || 0 != memcmp(fullPath, root, rootLength))
+    size_t pathLength = strlen(realFullPath);
+    if (pathLength < rootLength || 0 != memcmp(realFullPath, root, rootLength))
     {
+        
         // TODO(Mac): Add this message to PrjFSLib logging once available (#395)
-        LogError("GetRelativePath: root path '%s' is not a prefix of path '%s'\n", root, fullPath);
+        LogError("GetRelativePath: root path '%s' is not a prefix of path '%s'\n", root, realFullPath);
         return nullptr;
     }
     
-    const char* relativePath = fullPath + rootLength;
+    const char* relativePath = realFullPath + rootLength;
     if (relativePath[0] == '/')
     {
         relativePath++;
     }
     else if (rootLength > 0 && root[rootLength - 1] != '/' && pathLength > rootLength)
     {
         // TODO(Mac): Add this message to PrjFSLib logging once available (#395)
-        LogError("GetRelativePath: root path '%s' is not a parent directory of path '%s' (just a string prefix)\n", root, fullPath);
+        LogError("GetRelativePath: root path '%s' is not a parent directory of path '%s' (just a string prefix)\n", root, realFullPath);
```

Comment:
Same here, I think it would be useful to have both `fullPath` and `realFullPath`